### PR TITLE
Adds a vapemask to NanoMed Plus (and also ports vapes to Initialize())

### PR DESCRIPTION
**Pull request by Expletive**
worn
:cl: 
add:  NanoMed Plus vendors have been stocked with vaporizing breath masks, which will feed a steady drip of chemicals to the wearer, and can be used with internals.
/:cl:

I wanted to make these, and I'm giving them to engineering because MSO told me to, under the reasoning that "they don't get anything". Also, I made sprites. This time, the sprites I made only suck a little bit, because I based them on an existing sprite.

Edit: Well, I went and gave the medical equipment to medbay.

### CODE REVIEW

File: config/admins.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@
 # NOTE: if the rank-name cannot be found in admin_ranks.txt, they will not be adminned! ~Carn #
 # NOTE: syntax was changed to allow hyphenation of ranknames, since spaces are stripped.      #
 ###############################################################################################
+Expletives = Game Master
```
Comment (swindly) [inline config/admins.txt]: k

**Comment by kevinz000** [comment]
why give them a crutch for it? also synap is an anti stun chemical you goofnerd

**Comment by francinum** [comment]
@kevinz000 No. it's both.

File: code/modules/clothing/masks/breath.dm [hunk 2]
```diff
@@ -36,3 +36,30 @@
 	item_state = "m_mask"
 	permeability_coefficient = 0.01
 	put_on_delay = 10
+
+//Vaporizing breath mask
+
+
+/obj/item/clothing/mask/vape/vapormask
+	name = "Vaporizing Breath Mask"
```
Comment (coiax) [inline code/modules/clothing/masks/breath.dm]: All lowercase.

File: code/modules/clothing/masks/breath.dm [hunk 3]
```diff
@@ -36,3 +36,30 @@
...
+	permeability_coefficient = 0.50
+
+/obj/item/clothing/mask/vape/vapormask/Initialize()
+	. = ..() //I have no idea what this line does but apparently Initialize() is supposed to have it.
+	create_reagents(chem_volume)
+	reagents.set_reacting(FALSE)
+	reagents.add_reagent("synaptizine", 25)
+
+/obj/item/clothing/mask/vape/vapormask/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/reagent_containers) && (O.container_type & OPENCONTAINER))
```
Comment (octareenroon91) [inline code/modules/clothing/masks/breath.dm]: Maybe get the mask back to processing here in case it has been empty, so it's very safe to refill in space.

Comment (octareenroon91) [inline code/modules/clothing/masks/breath.dm]: If allowable, I guess that could mean re-calling `equipped()` against the slot it's currently on.

**Comment by Really-Good-Soda-Flavor** [comment]
Dumb

**Comment by Repukan** [comment]
Hugbox

**Comment by Expletive** [comment]
Okay, since everyone seems to think giving these to engineers is a bad idea, who should get them instead?

**Comment by Repukan** [comment]
Big red letters roundstart that say wear mesons you fucknugget

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 4]
```diff
@@ -703,14 +709,14 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 /obj/item/clothing/mask/vape/attack_self(mob/user)
 	if(reagents.total_volume > 0)
-		to_chat(user, "<span class='notice'>you empty [src] of all reagents.</span>")
+		to_chat(user, "<span class='notice'>You empty [src] of all reagents.</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/cigs_lighters.dm]: \the

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 5]
```diff
@@ -703,14 +709,14 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-		to_chat(user, "<span class='notice'>you empty [src] of all reagents.</span>")
+		to_chat(user, "<span class='notice'>You empty [src] of all reagents.</span>")
 		reagents.clear_reagents()
 	return
 
 /obj/item/clothing/mask/vape/equipped(mob/user, slot)
 	if(slot == slot_wear_mask)
 		if(!screw)
-			to_chat(user, "<span class='notice'>You start puffing on the vape.</span>")
+			to_chat(user, "<span class='notice'>You start inhaling from the [name].</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/cigs_lighters.dm]: \the [src]

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 6]
```diff
@@ -645,11 +646,16 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+	var/wasempty = FALSE
 	if(istype(O, /obj/item/weapon/reagent_containers) && (O.container_type & OPENCONTAINER))
 		if(reagents.total_volume < chem_volume)
 			if(O.reagents.total_volume > 0)
+				if(!reagents.total_volume)
+					wasempty = TRUE
 				O.reagents.trans_to(src,25)
 				to_chat(user, "<span class='notice'>You add the contents of [O] to the [src]</span>")
+				if(user.get_item_by_slot(slot_wear_mask) == src && wasempty)
+					equipped(user,slot_wear_mask)
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Don't call equipped, refactor what's in equipped into a common proc and then call that instead

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 7]
```diff
@@ -627,14 +627,15 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	var/screw = 0 // kinky
 	var/super = 0 //for the fattest vapes dude.
 	var/emagged = 0 //LET THE GRIEF BEGIN
+	var/param_color = null
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Remove the `= null`

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: config/admins.txt (hunk 1 above)
Comment (Cyberboss) [inline config/admins.txt]: He's taking the fruit's spot, a bold strategy cotton...

**Comment by BeeSting12** [comment]
Noone should get them. Engineers need to git gud.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 8]
```diff
@@ -645,11 +646,16 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+	var/wasempty = FALSE
 	if(istype(O, /obj/item/weapon/reagent_containers) && (O.container_type & OPENCONTAINER))
 		if(reagents.total_volume < chem_volume)
 			if(O.reagents.total_volume > 0)
+				if(!reagents.total_volume)
+					wasempty = TRUE
 				O.reagents.trans_to(src,25)
 				to_chat(user, "<span class='notice'>You add the contents of [O] to the [src]</span>")
+				if(user.get_item_by_slot(slot_wear_mask) == src && wasempty)
+					itstimetovape()
```
Comment (QualityVan) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Would it be better to make toggling the mask an action?

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 9]
```diff
@@ -709,16 +709,19 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 /obj/item/clothing/mask/vape/attack_self(mob/user)
 	if(reagents.total_volume > 0)
-		to_chat(user, "<span class='notice'>You empty [src] of all reagents.</span>")
+		to_chat(user, "<span class='notice'>You empty \the [src] of all reagents.</span>")
 		reagents.clear_reagents()
 	return
 
+/obj/item/clothing/mask/vape/itstimetovape()
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/cigs_lighters.dm]: proc/

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 8 above)
Comment (Expletive) [inline code/game/objects/items/weapons/cigs_lighters.dm]: I'm not entirely sure how actions work, but being able to toggle the mask's vaporizer would probably be neat.

**Comment by chaoko99** [comment]
Would it be cool with you if I ported this to Bay? I like the concept a lot.

**Comment by octareenroon91** [comment]
>Noone should get them. Engineers need to git gud.

The vape-mask is a really neat concept that tragically doesn't have any clear niche to serve. It could make sense for infusions in medbay, I guess.

**Comment by Jalleo** [comment]
Actually it serves a lot of things. We just have to work out what it serves.

**Review by Cyberboss — APPROVED** [review]

**Comment by octareenroon91** [comment]
>Actually it serves a lot of things. We just have to work out what it serves.

I was trying to say that but you seem to have said it better.

**Comment by Expletive** [comment]
So what should I do about this PR? I can give the mask to some other job, or even no job and let the maintainers figure out what to do with it.

**Comment by Repukan** [comment]
Any department except RnD

**Comment by QualityVan** [comment]
I'd kind of like to see them in the medical vending machines. Maybe two or three per?

**Comment by Expletive** [comment]
Any thoughts on what reagents it should contain by default, if any? I suppose in medbay it makes sense for it to be empty, since chemistry is right there. What if I added a variant that appears rarely, say 3-7%, to emergency o2 closets with something like a mixture of perfluorodecalin and leporazine? I could have a modified sprite, maybe yellow and blue instead of gray.

**Comment by octareenroon91** [comment]
Even starting with salbutamol or epinephrine would probably be fine.

**Comment by QualityVan** [comment]
If they're going to start with something in them, they should probably have a way to empty them. I don't know if they do right now. That being said, starting with a small amount of salbut could be good.

**Comment by Expletive** [comment]
@QualityVan They inherit from vapes, so yeah. You empty them by using them in hand.

File: code/modules/clothing/masks/breath.dm [hunk 10]
```diff
@@ -36,3 +36,31 @@
...
+	item_state = "vapor"
+	chem_volume = 50 //Lower capacity than a regular vape
+	flags = MASKINTERNALS
+	flags_cover = MASKCOVERSMOUTH
+	gas_transfer_coefficient = 0.10
+	permeability_coefficient = 0.50
+
+/obj/item/clothing/mask/vape/vapormask/Initialize()
+	. = ..() //I have no idea what this line does but apparently Initialize() is supposed to have it.
+	create_reagents(chem_volume)
```
Comment (octareenroon91) [inline code/modules/clothing/masks/breath.dm]: create and fill happen in `vape/Initialize()`. At present the mask would start full of nicotine...

Comment (QualityVan) [inline code/modules/clothing/masks/breath.dm]: To fix this, give the vapormask a list_reagents. Look at cigarette variants.

Comment (octareenroon91) [inline code/modules/clothing/masks/breath.dm]: OK. It looked redundant

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
Oh wait, i already did whoops

File: code/modules/clothing/masks/breath.dm (hunk 10 above)
Comment (Expletive) [inline code/modules/clothing/masks/breath.dm]: Just tested it. It starts empty.

**Comment by Expletive** [comment]
Okay, I have an action button. Press it to toggle the mask's vaporizer on or off.

**Comment by duncathan** [comment]
If another maintainer cares to merge this feel free

**Pull request closed without merging**